apps/prairielearn/src/question-servers/freeform.js

```diff
@@ -549,13 +549,23 @@ async function experimentalProcess(phase, codeCaller, data, context, html) {
 
   return {
     courseIssues,
-    data: result?.data ?? data,
+    // Casting to the type of the argument is safe; a given phase is never allowed
```

Comment:
I would prefer to cast `result.data` rather than cast this expression, as data is already guaranteed to be this type. Maybe typing result with `unknowns` + `result.data: T`

Reply:
Done in 50f06bd.